app/lib/question_navigation.rb

```diff
@@ -163,4 +163,30 @@ class QuestionNavigation
     Questions::SuccessfullySubmittedController,
     Questions::FeedbackController,
   ].freeze
+
+  def self.determine_current_question(intake)
+    return nil if intake.completed_at?
+    return Questions::ConsentController.to_path_helper unless intake.primary_consented_to_service_at?
+
+    if intake.completed_yes_no_questions_at? && intake.document_types_definitely_needed.present?
+      return Documents::OverviewController.to_path_helper
+    end
+
+    # If yes/no questions completed + docs uploaded, start at InterviewSscheduling. Else, start at consent
+    i = intake.completed_yes_no_questions_at? ? FLOW.index(Questions::OverviewDocumentsController) : FLOW.index(Questions::ConsentController)
+    found_path = nil
+    while found_path.nil?
+      i += 1
+      next if i == FLOW.index(Questions::OverviewDocumentsController) # we handle documents seperately above
+
+      question = FLOW[i]
+      next unless question.show?(intake)
+
+      answer = intake.send(question.form_class.attribute_names.first)
+      next unless ["unfilled", nil].include? answer
+
+      found_path = question.to_path_helper # otherwise, this is the found_path
+    end
+    found_path.to_s
```

Comment:
i actually find this whole function rather difficult to read and still don't understand what it's doing in all of the various cases

Reply:
I'm happy to take specific or vague  suggestions of how to improve its readability. It's a utility function to determine the question the client hasn't yet answered. What it's doing is looping through every question form to see which question the client hasn't yet answered and set that as their current question.

Reply:
This loop examines the flow & answers in a way that should handle most of the situations I can think of, but I'm not sure if there are edge cases we should be considering. I don't feel confident in my ability to assess it reliably. I'd prefer something simpler if possible.

Reply:
Would also still prefer most of this to live in a backfill rake task. It feels like permanent complexity for a one-time need.